File: API/IRemoteService.cs
using System;
using ACRemote;

namespace ACRemote.API {
	public interface IRemoteService {
		void set_power(bool val);
		bool get_power();
		void set_temp(int val);
		int get_temp();
		void set_mode(string mode);
		string get_mode();
		void set_speed(string speed);
		string get_speed();
		double actual_temp();
		double actual_humidity();
	}
}

File: API/IRemoteService.vb
' Interface for JSON service
' Author: Kevin Ross

Namespace ACRemote.API
    Public Interface IRemoteService
        Sub set_power(val As Boolean)
        Function get_power() As Boolean
        Sub set_temp(val As Integer)
        Function get_temp() As Integer
        Sub set_mode(mode As String)
        Function get_mode() As String
        Sub set_speed(speed As String)
        Function get_speed() As String
        Function actual_temp() As Double
        Function actual_humidity() As Double
        Sub __set_temp(val As Double)
        Sub __set_humidity(val As Double)
        Sub reset()
    End Interface
End Namespace

